zrml/prediction-markets/src/lib.rs

```diff
@@ -1447,13 +1571,177 @@ mod pallet {
+                                            // AdvisoryBond was already returned when the market
+                                            // was approved. Approval is inevitable to reach this.
+                                            CurrencyOf::<T>::unreserve_named(
+                                                &RESERVE_ID,
+                                                &m.creator,
+                                                T::OracleBond::get(),
+                                            );
+                                        }
+
+                                        total_weight = total_weight
+                                            .saturating_add(dbweight.reads(2))
+                                            .saturating_add(dbweight.writes(2));
+                                        Ok(())
+                                    });
+
+                                if let Err(err) = market_result {
+                                    log::error!(
+                                        "[PredictionMarkets] Cannot find market associated to \
+                                         market id.
+                                    market_id: {:?}, error: {:?}",
+                                        subsidy_info.market_id,
+                                        err
+                                    );
+                                    return true;
+                                }
+
+                                let _ =
+                                    T::MarketCommons::remove_market_pool(&subsidy_info.market_id);
+                                total_weight =
+                                    total_weight.saturating_add(one_read).saturating_add(one_write);
+                                Self::deposit_event(Event::MarketInsufficientSubsidy(
+                                    subsidy_info.market_id,
+                                ));
+                            }
+
+                            return false;
+                        } else if let Err(err) = end_subsidy_result {
+                            log::error!(
+                                "[PredictionMarkets] An error occured during end of subsidy phase.
+                        pool_id: {:?}, market_id: {:?}, error: {:?}",
+                                pool_id,
+                                subsidy_info.market_id,
+                                err
+                            );
+                        }
+                    }
+
+                    return true;
+                };
+
+            let mut weight_basis = 0;
+            <MarketsCollectingSubsidy<T>>::mutate(
+                |e: &mut Vec<SubsidyUntil<T::BlockNumber, MomentOf<T>, MarketIdOf<T>>>| {
+                    weight_basis = T::WeightInfo::process_subsidy_collecting_markets_raw(
+                        e.len().saturated_into(),
+                    );
+                    e.retain(retain_closure);
```

Comment:
**Nit**

`retain` is a bit tricky. Since errors are omitted, it will be almost impossible to assert the integrity of an operation

Reply:
Will be improved in a future iteration. I didn't want to return with an error, since all other (Rikiddo) markets would not be processed in that case. Instead, the error-producing market will be kept and on every block error message will indicate that something went wrong, which we can fix then during the next update.

Reply:
Sure